test/syslogger_SUITE.erl

```diff
@@ -0,0 +1,151 @@
+-module(syslogger_SUITE).
+
+-export([all/0,
+         init_per_suite/1,
+         end_per_suite/1,
+         init_per_testcase/2,
+         end_per_testcase/2]).
+
+-export([
+    syslogger_is_used_to_log/1,
+    log_at_every_level/1,
+    log_at_custom_level/1
+]).
+
+-include_lib("common_test/include/ct.hrl").
+-include_lib("eunit/include/eunit.hrl").
+
+-define(HANDLER_ID, sys_log).
+-define(IDENT, "mongooseim").
+-define(FACILITY, local0).
+
+all() ->
+    [syslogger_is_used_to_log,
+     log_at_every_level,
+     log_at_custom_level
+    ].
+
+init_per_suite(Config) ->
+    LoggerConfig = logger:get_primary_config(),
+    logger:set_primary_config(level, info),
+
+    setup_meck(Config),
+    ok = mongoose_logger_running(),
+
+    [{logger_primary_config, LoggerConfig} | Config].
+
+end_per_suite(Config) ->
+    logger:remove_handler(?HANDLER_ID),
+    logger:set_primary_config(?config(logger_primary_config, Config)),
+
+    unload_meck(Config),
+    ok.
+
+init_per_testcase(_TC, Config) ->
+    meck:reset(syslogger),
+    Config.
+
+end_per_testcase(_TC, _Config) ->
+    ok.
+
+%%
+%% Tests
+%%
+syslogger_is_used_to_log(_Config) ->
+    logger:log(error, "Something"),
+
+    ?assert(meck:validate(syslogger)),
+    ok = meck:wait(syslogger, log, [#{level => error,
+                                           msg => {string, "Something"},
+                                           meta => '_'},
+        #{id => ?HANDLER_ID, ident => ?IDENT, facility => ?FACILITY}],
+        timer:seconds(1)).
+
+log_at_every_level(_Config) ->
+    %% given
+    [ begin
+      %% when
+          mongoose_logs:set_global_loglevel(LName),
+          %% then
+          log_at_level(LName)
+      end || {_, LName} <- levels() ].
+
+log_at_level(none) ->
+    %% When log level `none` is set and we log on each possible level...
+    [ logger:log(LevelName, "", []) || {_, LevelName} <- levels(), LevelName /= none ],
+    %% ...then nothing ends up in the log file.
+    %% (polling doesn't make sense in this one case, we have to sleep)
```

Comment:
By the way, this comment doesn't make sense, is it a leftover from the times when there was a timer:sleep here? If so, you could remove it for this testsuite and the other one 😅